### PR TITLE
Add coffee script coding style

### PR DESCRIPTION
**Pull request by minsikzzang**
Task
- [x] Add initial draft of coffee script style guide

### CODE REVIEW

File: coding-styles/coffeescript/README.md [hunk 1]
```diff
@@ -0,0 +1,759 @@
...
+
+    # good
+    if a and b
+    ```
+
+  - [17.2](#17.2)<a name='17.2'></a> or is preferred over ||.
+
+    ```coffeescript
+    # bad
+    if a && b
```
Comment (ericescalante) [inline coding-styles/coffeescript/README.md]: && should be ||

File: coding-styles/coffeescript/README.md [hunk 2]
```diff
@@ -0,0 +1,759 @@
...
+
+    # good
+    if a and b
```
Comment (ericescalante) [inline coding-styles/coffeescript/README.md]: 'and' should be 'or'

File: coding-styles/coffeescript/README.md [hunk 3]
```diff
@@ -0,0 +1,759 @@
...
+
+  - [17.5](#17.5)<a name='17.5'></a> not is preferred over !.
+
+    ```coffeescript
+    # bad
+    if not b
+
+    # good
+    if !b
+    ```
```
Comment (ericescalante) [inline coding-styles/coffeescript/README.md:700]: this needs switching

**Comment by ericescalante** [comment]
I would also add an entry about using => over a -> with a explicit bind for 'this'
see:
http://coffeescript.org/#fat-arrow

File: coding-styles/coffeescript/README.md (hunk 2 above)
Comment (minsikzzang) [inline coding-styles/coffeescript/README.md]: I've changed my commit, but github seems doing bit crazy now :)

**Comment by ericescalante** [comment]
:+1: Rock n' roll!

**Comment by willhall88** [comment]
:+1:

**Pull request merged**